inc/admin.php

```diff
@@ -469,3 +472,42 @@ function admin_notices() {
 	endif;
 }
 add_action( 'admin_notices', __NAMESPACE__ . '\admin_notices' );
+
+/**
+ * Clean orphaned posts if found
+ *
+ * @param int    $option    adstxt | app_adstxt post id
+ * @param string $post_type post type
+ *
+ * @return void|boolean
+ */
+function clean_orphaned_posts( $option, $post_type ) {
+	if ( ! $option || ! $post_type ) {
+		return false;
+	}
+
+	$args = [
+		'fields'    => 'ids', // Only get post IDs
+		'post_type' => $post_type,
+	];
+
+	$ads_posts = get_posts( $args );
+
+	if ( empty( $ads_posts ) ) {
+		return false;
+	}
+
+	// Search and remove the element using unset()
+	$index = array_search( $option, $ads_posts, true );
```

Comment:
WP always stores options as a string so casting is needed for a strict type comparison.

At the moment, the post is created and then deleted on each page load.

```suggestion
	$index = array_search( (int) $option, $ads_posts, true );
```